### ISSUE TITLE
URI missing errors in GTK UI

### ISSUE THREAD

**Issue by Mte90** [quexten/goldwarden#111]
It's me again and I was able to run the ui but there are various errors and deprecated errors:
```
S daemon running True
autostart enabled..!?
(':1.25', '/org/freedesktop/portal/desktop/request/1_629/com/quexten/Goldwarden/18088377', 'org.freedesktop.portal.Request', 'Response', GLib.Variant('(ua{sv})', (0, {'background': <true>, 'autostart': <true>})), None)
/home/mte90/Desktop/kde/goldwarden/ui/./settings.py:50: DeprecationWarning: Gtk.Widget.get_style_context is deprecated
  self.autotype_button.get_style_context().add_class("suggested-action")
/home/mte90/Desktop/kde/goldwarden/ui/./settings.py:50: DeprecationWarning: Gtk.StyleContext.add_class is deprecated
  self.autotype_button.get_style_context().add_class("suggested-action")
/home/mte90/Desktop/kde/goldwarden/ui/./settings.py:149: DeprecationWarning: Adw.ActionRow.set_icon_name is deprecated
  self.last_sync_row.set_icon_name("emblem-synchronizing-symbolic")
/home/mte90/Desktop/kde/goldwarden/ui/./settings.py:289: DeprecationWarning: Gtk.Dialog.get_content_area is deprecated
  dialog.get_content_area().append(auth_preference_group)
/home/mte90/Desktop/kde/goldwarden/ui/./settings.py:306: DeprecationWarning: Gtk.Dialog.add_button is deprecated
  dialog.add_button("Login", Gtk.ResponseType.OK)
Gtk-Message: 16:19:55.247: GtkDialog mapped without a transient parent. This is discouraged.
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:6: PyGIWarning: Notify was imported without specifying a version first. Use gi.require_version('Notify', '0.7') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, Adw, GLib, Notify, Gdk
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:84: DeprecationWarning: Gtk.Widget.hide is deprecated
  self.history_list.hide()
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:130: DeprecationWarning: Gtk.Widget.get_style_context is deprecated
  self.history_list.get_style_context().add_class("boxed-list")
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:130: DeprecationWarning: Gtk.StyleContext.add_class is deprecated
  self.history_list.get_style_context().add_class("boxed-list")
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:65: DeprecationWarning: Adw.ActionRow.set_icon_name is deprecated
  action_row.set_icon_name("dialog-password")
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:47: DeprecationWarning: Gtk.Widget.show is deprecated
  self.history_list.show()
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:6: PyGIWarning: Notify was imported without specifying a version first. Use gi.require_version('Notify', '0.7') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, Adw, GLib, Notify, Gdk
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:84: DeprecationWarning: Gtk.Widget.hide is deprecated
  self.history_list.hide()
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:130: DeprecationWarning: Gtk.Widget.get_style_context is deprecated
  self.history_list.get_style_context().add_class("boxed-list")
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:130: DeprecationWarning: Gtk.StyleContext.add_class is deprecated
  self.history_list.get_style_context().add_class("boxed-list")
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:65: DeprecationWarning: Adw.ActionRow.set_icon_name is deprecated
  action_row.set_icon_name("dialog-password")
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py:47: DeprecationWarning: Gtk.Widget.show is deprecated
  self.history_list.show()
Traceback (most recent call last):
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 75, in <lambda>
    self.text_view.connect("changed", lambda entry: on_type(entry))
                                                    ^^^^^^^^^^^^^^
  File "/home/mte90/Desktop/kde/goldwarden/ui/./autofill.py", line 70, in on_type
    action_row.uri = i["uri"]
                     ~^^^^^^^
KeyError: 'uri'
```

The window that open to search require for the user password, a window to accept this access and when I search doesn't find anything.

**Comment by Mte90** [comment]
A check in the code just showed that `uri` is not defined everywhere but your code expect it and crash.
Maybe add an alert to inform the user to add a uri to the voices that are missing can help.

![image](https://github.com/quexten/goldwarden/assets/403283/d92b3525-276f-4f57-92cd-4b2b0a6aaec3)

**Comment by Mte90** [comment]
So I checked on my vaultwarden instance also items that have a URL configured doesn't have the `uri` value the code is looking for, the output is just like in the screenshot above.

**Comment by Mte90** [comment]
Let me know how can I help you, also on patching the code etc

**Comment by quexten** [comment]
Might be fixed in 11ad8c4f48d128df93aa006890676b7262d198ee , beware though that the master branch might have changed significantly since you cloned it, mainly in the structure of the gtk ui code.
